include/aws/common/json.h

```diff
@@ -23,13 +23,25 @@ AWS_EXTERN_C_BEGIN
  *
  * Note: You will need to free the memory for the aws_json_value using aws_json_destroy on the aws_json_value or
  * on the object/array containing the aws_json_value.
- * @param string A byte pointer to the string you want to store in the aws_json_value
+ * @param string A byte cursor you want to store in the aws_json_value
```

Comment:
added a generic comment that it can be slower. we try to hide implementation details, since this header is a kind of interface. but in all likely hood we are not going to change underlying impl any time soon